transmart-core-db/grails-app/services/org/transmartproject/db/ontology/MDStudiesService.groovy

```diff
@@ -106,23 +108,34 @@ class MDStudiesService implements MDStudiesResource, ApplicationRunner {
         study
     }
 
+    @Override
+    MDStudy getStudyByStudyIdForUser(String studyId, User currentUser) throws NoSuchResourceException {
+        def user = usersResource.getUserFromUsername(currentUser.username)
+        def study = Study.findByStudyId(studyId)
+        if (isLegacyStudy(study)) {
+            study = null
+        }
+        if (study == null || !user.canPerform(ProtectedOperation.WellKnownOperations.READ, study)) {
+            throw new AccessDeniedException("Access denied to study or study does not exist: ${studyId}")
+        }
+        study.dimensions.size()
+        study
+    }
```

Comment:
This looks like unnecessary code duplication. Either change the clients to use `getStudyByStudyIdsForUser([studyId], user)[0]` or change the implementation of this to delegate to the new method in the same way.

By the way, this is an interesting case study for you code quality goal. You might want to consider if/why you didn't catch this yourself and how you might prevent that in the future.

Reply:
Whether to change the implementation of `getStudyByStudyIdForUser` or to change the clients is a design decision with (imo) not a clear answer. It's a trade off between simplicity in the MDStudiesService interface and simplicity in the clients. Though if there are more than two clients (or expected to be soon), I would opt for keeping the interface method and delegating the implementation.

Reply:
These lines are not actually a part of my PR, I just restored a method that I removed in my previous commit (getStudyByStudyIdForUser), so this duplication is already part of the main code. But I agree, I should have noticed that and improve it. I will add a change here.

Reply:
According to github, this PR introduces the `getStudiesByStudyIdsForUser` call, thus introducing the duplication :). But I can understand you didn't notice it over the separate commits.

Reply:
the duplication is mainly between `getStudyForUser` and `getStudyByStudyIdForUser`, in `getStudiesByStudyIdsForUser` the implementation is a bit different, because it does not throw an error, when a user has no access or for legacy studies (just does not include these studies in the results). Should this be changed?